### PR TITLE
feat: support to track written size

### PR DESCRIPTION
**Pull request by ZENOTME**
As mention in #72, this PR implement a write wrapper to track the written size.

### CODE REVIEW

File: src/io/parquet/write.rs [hunk 1]
```diff
@@ -1,11 +1,68 @@
...
 use parquet::arrow::AsyncArrowWriter;
 use parquet::file::properties::WriterProperties;
+use tokio::io::AsyncWrite;
 
 use crate::Result;
 
+/// `TrackWriter` is used to track the written size.
+struct TrackWriter {
+    writer: Writer,
+    written_size: Arc<AtomicUsize>,
```
[the remark this answers is not in the document]
Comment (ZENOTME) [reply src/io/parquet/write.rs:17]: We need to expose it to observe it externally, e.g.
```
let (writer,written_size) = build();
writer.write()
println!("{}",written_size);
writer.write()
println!("{}",written_size);
```
To achive this, we have to wrapper it with Arc and AtomicUsize for modify it🤔
How to achive it with usize?

File: src/io/parquet/write.rs [hunk 2]
```diff
@@ -50,23 +107,22 @@ impl ParquetWriterBuilder {
...
-    pub fn build(self) -> Result<ParquetWriter> {
-        let writer = AsyncArrowWriter::try_new(
-            self.writer,
-            self.arrow_schema,
-            self.buffer_size,
-            self.props,
-        )?;
-
-        Ok(ParquetWriter { writer })
+    pub fn build(self) -> Result<(ParquetWriter, Arc<AtomicUsize>)> {
```
Comment (Xuanwo) [inline src/io/parquet/write.rs:110]: I don't like the idea to return this in build. I prefer to change the API of `write()` to return the `(num_rows, written_size)`. But we can move on first to build our first table write PoC and polish here later.

File: src/io/parquet/write.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZENOTME) [reply src/io/parquet/write.rs:17]: The reason why we want to expose it is to used in 
https://github.com/icelake-io/icelake/blob/5659276d3bacefc37e9452724a8d463e1d6429fb/src/io/data_file_writer.rs#L27

For now, we will split data into multiple data file accoring to target_file_row_num. But I think that may cause a problem: We don't know how large a row will be, split accoring to target_file_row_num may caused the variant size of data file. Use `target_file_size` can avoid this problem.

So we can expose it to used in `target_file_size` first and refactor it when using metrics.🤔

[the remark this answers is not in the document]
Comment (Xuanwo) [reply src/io/parquet/write.rs:17]: > So we can expose it to used in `target_file_size` first and refactor it when using metrics.🤔

I don't think metrics is good to use here. The size of the parquet file is unknown until it is `close`, so the `target_file_size` parameter serves as a suggestion.

[the remark this answers is not in the document]
Comment (Xuanwo) [reply src/io/parquet/write.rs:17]: By the way, I don't think it's a critical issue. We can continue with our current tasks and deal with this issue at a later time.

File: src/io/parquet/write.rs (hunk 2 above)
Comment (ZENOTME) [reply src/io/parquet/write.rs:110]: > Please note that the size is incorrect until we call close (data could be still in buffer).

Yes, AsyncArrowWriter will buffer the data. So when we call write(), it don't means that the data will be writed.
So for `write(&record_batch)->(num_rows,writtern_size)`, Dose the writtern_size means size of record_batch? or the writtern_size return by writer?

Comment (Xuanwo) [reply src/io/parquet/write.rs:110]: > the writtern_size return by writer

We still need to wrap the writer, but don't need to expose to users.

File: src/io/parquet/write.rs [hunk 3]
```diff
@@ -84,8 +144,17 @@ impl ParquetWriter {
     /// # Note
     ///
     /// This function must be called before complete the write process.
-    pub async fn close(self) -> Result<FileMetaData> {
-        Ok(self.writer.close().await?)
+    pub async fn close(self) -> Result<(FileMetaData, usize)> {
```
Comment (Xuanwo) [inline src/io/parquet/write.rs:147]: Please always use `u64` for file size.

**Review by Xuanwo — APPROVED** [review]
Thanks!

**Pull request merged**